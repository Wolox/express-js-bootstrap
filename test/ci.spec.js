const utils = require('./helpers/utils'),
  { mockCommand } = require('./helpers/mocks'),
  { basicFiles, jenkinsFiles, travisFiles, examplePrompts } = require('./helpers/constants');

beforeAll(() => mockCommand());

const ciOptions = [['travis', travisFiles], ['jenkins', jenkinsFiles]];

describe.each(ciOptions)('%s project', (ciName, files) => {
  beforeAll(() =>
    utils.runKickoff({
      ...examplePrompts,
      projectName: 'CIProject',
      ci: ciName
    })
  );

  test(`creates files for ${ciName} project`, () => {
    utils.checkExistentFiles([basicFiles, files], 'CIProject');
  });

  test.each(files)('creates expected %s', file => {
    expect(utils.getFileContent(`CIProject/${file}`)).toMatchSnapshot();
  });
});
