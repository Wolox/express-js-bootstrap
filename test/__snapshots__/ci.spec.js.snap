// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jenkins project creates expected .woloxci/Dockerfile 1`] = `
"FROM node:8.9.12

WORKDIR /usr/src/app

ENV NODE_ENV testing

ENV HOME /usr/src/app

ENV BABEL_DISABLE_CACHE 1

RUN mkdir -p /install

ENV NODE_PATH=/install/node_modules

# Install app dependencies
# A wildcard is used to ensure both package.json AND package-lock.json are copied
# where available (npm@5+)
COPY package*.json /install/

WORKDIR /install

RUN npm install
RUN npm install -g gulp
# If you are building your code for production
# RUN npm install --only=production

RUN chmod a+r /usr/src/app

WORKDIR /usr/src/app

# Bundle app source
COPY . .
"
`;

exports[`jenkins project creates expected .woloxci/config.yml 1`] = `
"config:
  dockerfile: .woloxci/Dockerfile
  project_name: CIProject

steps:
  copy_node_modules:
    - cp -r $NODE_PATH/ ./
  lint:
    - npm run lint
  test:
    - npm run test

environment:
  GIT_COMMITTER_NAME: a
  GIT_COMMITTER_EMAIL: b
  LANG: C.UTF-8
"
`;

exports[`jenkins project creates expected Jenkinsfile 1`] = `
"@Library('wolox-ci') _

node {

  checkout scm

  woloxCi('.woloxci/config.yml');
}
"
`;

exports[`travis project creates expected .travis.yml 1`] = `
"language: node_js

node_js:
  - \\"8.9.12\\"

sudo: true

env:
  - CXX=g++-4.8 NODE_ENV=testing
addons:
  apt:
    sources:
    - ubuntu-toolchain-r-test
    packages:
    - g++-4.8
"
`;
